scripts/install_prerequisites.sh

```diff
@@ -42,102 +44,171 @@ function create_llvm_symlinks {
   wget https://github.com/Kitware/CMake/releases/download/v3.26.4/cmake-3.26.4-linux-$(uname -m).sh -O cmake-install.sh
   bash cmake-install.sh --skip-licence --exclude-subdir --prefix=/usr/local
 fi
 if [ ! -x "$(command -v ninja)" ]; then
-  temp_install_if_command_unknown unzip unzip
-  wget https://github.com/ninja-build/ninja/releases/download/v1.11.1/ninja-linux.zip
-  unzip ninja-linux.zip && mv ninja /usr/local/bin/ && rm -rf ninja-linux.zip
-fi
-if [ "$CC" == "" ] && [ "$CXX" == "" ]; then
-  source "$this_file_dir/install_toolchain.sh" -t gcc12
-fi
-
-llvm_dir="$LLVM_INSTALL_PREFIX/lib/cmake/llvm"
-if [ ! -d "$llvm_dir" ]; then
-  echo "Could not find llvm libraries."
-
-  if [ ! -d "$PYBIND11_INSTALL_PREFIX" ]; then
-    echo "Building PyBind11..."
-    repo_root="$(git rev-parse --show-toplevel)" && cd "$repo_root"
-    git submodule update --init --recursive --recommend-shallow --single-branch tpls/pybind11 && cd -
-    mkdir "$repo_root/tpls/pybind11/build" && cd "$repo_root/tpls/pybind11/build"
-    cmake -G Ninja ../ -DCMAKE_INSTALL_PREFIX="$PYBIND11_INSTALL_PREFIX"
-    cmake --build . --target install --config Release && cd -
-  fi
-
-  # Build llvm libraries from source and install them in the install directory
-  set +e && source "$this_file_dir/build_llvm.sh" -v && set -e
-  (return 0 2>/dev/null) && is_sourced=true || is_sourced=false
+  echo "Installing Ninja..."
+  temp_install_if_command_unknown wget wget
+  temp_install_if_command_unknown make make
 
-  if [ ! -d "$llvm_dir" ]; then
-    echo "Failed to find directory $llvm_dir."
-    if $is_sourced; then return 1; else exit 1; fi
-  fi
-else 
-  echo "Configured C compiler: $CC"
-  echo "Configured C++ compiler: $CXX"
+  # The pre-built binary for Linux on GitHub is built for x86_64 only, 
+  # see also https://github.com/ninja-build/ninja/issues/2284.
+  wget https://github.com/ninja-build/ninja/archive/refs/tags/v1.11.1.tar.gz
+  tar -xzvf v1.11.1.tar.gz && cd ninja-1.11.1
+  cmake -B build && cmake --build build
+  mv build/ninja /usr/local/bin/
+  rm -rf v1.11.1.tar.gz ninja-1.11.1
 fi
 
+# [Blas] Needed for certain optimizers
 if [ ! -f "$BLAS_INSTALL_PREFIX/libblas.a" ] && [ ! -f "$BLAS_INSTALL_PREFIX/lib/libblas.a" ]; then
-  if [ -x "$(command -v apt-get)" ]; then
-    apt-get update
-  fi
-
+  echo "Installing BLAS..."
   temp_install_if_command_unknown wget wget
   temp_install_if_command_unknown make make
   if [ ! -x "$(command -v "$FC")" ]; then
     temp_install_if_command_unknown gcc gcc
     temp_install_if_command_unknown g++ g++
     temp_install_if_command_unknown gfortran gfortran
+  elif [ ! -x "gfortran" ]; then
+    ln -s "$FC" /usr/bin/gfortran
   fi
 
   # See also: https://github.com/NVIDIA/cuda-quantum/issues/452
   wget http://www.netlib.org/blas/blas-3.11.0.tgz
-  tar -xzvf blas-3.11.0.tgz && cd BLAS-3.11.0
-  make && mkdir -p "$BLAS_INSTALL_PREFIX" && mv blas_LINUX.a "$BLAS_INSTALL_PREFIX/libblas.a"
+  tar -xzvf blas-3.11.0.tgz 
```

Comment:
Suggestion - can we capture the exact version numbers in a variable (similar to how its implemented in `configure_build.sh`), so, update and maintenance is easier?